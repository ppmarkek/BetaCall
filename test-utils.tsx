import { ReactElement, FC } from 'react';
import { ChakraProvider, defaultSystem } from '@chakra-ui/react';
import { render, RenderOptions } from '@testing-library/react';

const Wrapper: FC<{ children: React.ReactNode }> = ({ children }) => (
  <ChakraProvider value={defaultSystem}>{children}</ChakraProvider>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';
export { customRender as render };
