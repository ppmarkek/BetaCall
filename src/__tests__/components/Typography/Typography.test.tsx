import React from 'react';
import { render } from '../../../../test-utils';
import '@testing-library/jest-dom';
import Typography from '@/components/typography/typography';

describe('Typography Component', () => {
  describe('typography with variant Button', () => {
    it('renders the typography with fontWeight Bold', () => {
      const { container } = render(
        <Typography variant="Button" weight="Bold">
          Test
        </Typography>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders the typography with fontWeight Regular', () => {
      const { container } = render(
        <Typography variant="Button" weight="Regular">
          Test
        </Typography>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders the typography with fontWeight Small', () => {
      const { container } = render(
        <Typography variant="Button" weight="Small">
          Test
        </Typography>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders the typography with fontWeight Default', () => {
      const { container } = render(
        <Typography variant="Button">Test</Typography>
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe('typography with variant Regular', () => {
    it('renders the typography with fontWeight Bold', () => {
      const { container } = render(
        <Typography variant="Regular" weight="Bold">
          Test
        </Typography>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders the typography with fontWeight Regular', () => {
      const { container } = render(
        <Typography variant="Regular" weight="Regular">
          Test
        </Typography>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders the typography with fontWeight Small', () => {
      const { container } = render(
        <Typography variant="Regular" weight="Small">
          Test
        </Typography>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders the typography with fontWeight Default', () => {
      const { container } = render(
        <Typography variant="Regular">Test</Typography>
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe('typography with variant SmallRegular', () => {
    it('renders the typography with fontWeight Bold', () => {
      const { container } = render(
        <Typography variant="SmallRegular" weight="Bold">
          Test
        </Typography>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders the typography with fontWeight Regular', () => {
      const { container } = render(
        <Typography variant="SmallRegular" weight="Regular">
          Test
        </Typography>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders the typography with fontWeight Small', () => {
      const { container } = render(
        <Typography variant="SmallRegular" weight="Small">
          Test
        </Typography>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders the typography with fontWeight Default', () => {
      const { container } = render(
        <Typography variant="SmallRegular">Test</Typography>
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe('typography with variant Default', () => {
    it('renders the typography with fontWeight Bold', () => {
      const { container } = render(
        <Typography weight="Bold">
          Test
        </Typography>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders the typography with fontWeight Regular', () => {
      const { container } = render(
        <Typography weight="Regular">
          Test
        </Typography>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders the typography with fontWeight Small', () => {
      const { container } = render(
        <Typography weight="Small">
          Test
        </Typography>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders the typography with fontWeight Default', () => {
      const { container } = render(
        <Typography>Test</Typography>
      );
      expect(container).toMatchSnapshot();
    });
  });
});
