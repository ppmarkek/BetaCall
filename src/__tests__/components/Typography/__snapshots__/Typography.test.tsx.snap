// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Component typography with variant Button renders the typography with fontWeight Bold 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 16px;
  font-weight: 700;
  line-height: 19px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant Button renders the typography with fontWeight Default 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 16px;
  font-weight: 400;
  line-height: 19px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant Button renders the typography with fontWeight Regular 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 16px;
  font-weight: 400;
  line-height: 19px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant Button renders the typography with fontWeight Small 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 16px;
  font-weight: 300;
  line-height: 19px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant Default renders the typography with fontWeight Bold 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 14px;
  font-weight: 700;
  line-height: 21px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant Default renders the typography with fontWeight Default 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 14px;
  font-weight: 400;
  line-height: 21px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant Default renders the typography with fontWeight Regular 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 14px;
  font-weight: 400;
  line-height: 21px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant Default renders the typography with fontWeight Small 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 14px;
  font-weight: 300;
  line-height: 21px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant Regular renders the typography with fontWeight Bold 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 14px;
  font-weight: 700;
  line-height: 21px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant Regular renders the typography with fontWeight Default 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 14px;
  font-weight: 400;
  line-height: 21px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant Regular renders the typography with fontWeight Regular 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 14px;
  font-weight: 400;
  line-height: 21px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant Regular renders the typography with fontWeight Small 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 14px;
  font-weight: 300;
  line-height: 21px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant SmallRegular renders the typography with fontWeight Bold 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 12px;
  font-weight: 700;
  line-height: 18px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant SmallRegular renders the typography with fontWeight Default 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 12px;
  font-weight: 400;
  line-height: 18px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant SmallRegular renders the typography with fontWeight Regular 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 12px;
  font-weight: 400;
  line-height: 18px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;

exports[`Typography Component typography with variant SmallRegular renders the typography with fontWeight Small 1`] = `
.emotion-0 {
  font-family: Lato;
  font-size: 12px;
  font-weight: 300;
  line-height: 18px;
}

<div>
  <p
    class="emotion-0"
  >
    Test
  </p>
</div>
`;
